mrc-4924: Move new project actions under a single button

r-ash opened the pull request:
## Description

This PR will
* For logged in users, move the new project and upload from zip actions into a button on the projects page
* For guest users, move the upload from zip into step 1

This is in preparation for adding an upload output zip from ADR action under the new button.

## Type of version change

Patches

## Checklist

- [x] I have incremented version number, or version needs no increment
- [x] The build passed successfully, or failed because of ADR tests

M-Kusumgar reviewed (commented):
Looks great! just a couple of minor comments

Also unrelated bug, i was able to create two projects with the same name by creating a copy of the project!

EmmaLRussell reviewed (commented):
Works great!

The Projects page looks a bit desolate now when there aren't any projects yet. Feel like we could still have some sort of 'Create a new project to get started:' or 'Projects will appear here' text. 
![image (9)](https://github.com/mrc-ide/hint/assets/44669576/cb49722b-a004-4471-b4ce-58daba156c08)

I don't really like the "Phillips head screw" icon for 'Create project' in the drop down menu. You've already got the + icon (no circle) on the button, implying that it's relevant for all options, so I feel like it's a bit inconsistent to have that. It's also quite hard to read. I'd probably go for the plain + icon, or maybe the 'edit' icon which sometimes gets used for 'new file' operations. Or just do without the icons tbh, not sure they add much. 

Button maybe a bit oversized?

r-ash commented on src/app/static/src/app/components/header/UserHeader.vue:
```diff
@@ -5,13 +5,13 @@
                 <span v-if="!isGuest" class="pr-2 mr-2 border-right text-white">
                     <span v-translate="'loggedInAs'"></span> {{ user }}
                 </span>
+                <div v-else style="flex:auto"></div>
```
Using this in the header, because we no longer have the "File" button in the header and we may or may not have the user logged in text thought we needed some other div which would expand to fill the horizontal space. So using this to force the separation in the header between the "Naomi" text and the version number drop down.

r-ash commented on src/app/static/src/app/index.ts:
```diff
@@ -10,6 +10,7 @@ import 'floating-vue/dist/style.css';
 import "../scss/style.scss";
 import "bootstrap/scss/bootstrap-grid.scss";
 import "@reside-ic/vue-nested-multiselect/style.css";
+import "bootstrap/dist/js/bootstrap.bundle.js";
```
So without this the new project button doesn't actually open and show the drop down menu

r-ash replied:
Thanks! Just gave this another go and has simplified this a fair bit

r-ash commented:
> Works great!
> 
> The Projects page looks a bit desolate now when there aren't any projects yet. Feel like we could still have some sort of 'Create a new project to get started:' or 'Projects will appear here' text. ![image (9)](https://private-user-images.githubusercontent.com/44669576/305422529-cb49722b-a004-4471-b4ce-58daba156c08.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk2MzM1NjEsIm5iZiI6MTcwOTYzMzI2MSwicGF0aCI6Ii80NDY2OTU3Ni8zMDU0MjI1MjktY2I0OTcyMmItYTAwNC00NDcxLWI0Y2UtNThkYWJhMTU2YzA4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzA1VDEwMDc0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNhYmE4YWZlNGVhZWE3YWMxMjE0NmZjN2VkMzIwNzFkZjlmNjBhM2Y2NjQ3ZWNhMDZiYzM4Mzg2ZjUyNGQwMmQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.qnUh1FDfvWt8dMfTa60I14spAiIKYQemTwGGWz6d55U)
> 
> I don't really like the "Phillips head screw" icon for 'Create project' in the drop down menu. You've already got the + icon (no circle) on the button, implying that it's relevant for all options, so I feel like it's a bit inconsistent to have that. It's also quite hard to read. I'd probably go for the plain + icon, or maybe the 'edit' icon which sometimes gets used for 'new file' operations. Or just do without the icons tbh, not sure they add much.
> 
> Button maybe a bit oversized?

Thanks, I've added some placeholder text which will bw shown when there are no projects. Let me know what you think.

Updated the icon too, to a regular +. Deliberately went for an oversized button as I wanted to make it super obvious that this is the most common action used from the page

r-ash commented:
> Also unrelated bug, i was able to create two projects with the same name by creating a copy of the project!

Ticket for that https://mrc-ide.myjetbrains.com/youtrack/issue/mrc-4207/Cloning-a-project-should-not-allow-user-to-create-a-project-with-the-same-name

r-ash commented on src/app/static/src/app/components/uploadInputs/UploadInputs.vue:
```diff
@@ -224,10 +199,26 @@
+                getPlottingMetadata: 'metadata/getPlottingMetadata',
+                preparingRehydrate: 'load/preparingRehydrate',
+            }),
+            loadZip() {
+                const input = this.$refs.loadZip as HTMLInputElement;
+                if (input.files && input.files.length > 0) {
+                    const file = input.files[0];
+                    this.clearLoadZipInput();
+                    this.preparingRehydrate(getFormData(file));
+                }
+            },
+            clearLoadZipInput() {
+                // clearing value because browser does not
+                // allow selection of the same file twice
+                const input = this.$refs.loadZip as HTMLInputElement
+                input.value = ""
+            },
```
Thanks! Ticket here https://mrc-ide.myjetbrains.com/youtrack/issue/mrc-5145/Refactor-UploadInputs-to-composition-API

r-ash commented on src/app/static/package.json:
```diff
@@ -11,7 +11,7 @@
     "test": "vue-cli-service test:unit -c jest.config.js --maxWorkers=30%",
     "browser-test": "npx playwright test",
-    "integration-test": "jest --testPathIgnorePatterns adr-dataset -c jest.integration.config.js",
+    "integration-test": "jest --testPathIgnorePatterns adr-dataset -c jest.integration.config.js --maxWorkers=30%",
```
I was seeing intermittent failures on CI without this :(

EmmaLRussell reviewed (approved):
LGTM
